dogfood/main.tf

```diff
@@ -95,48 +95,75 @@ resource "coder_agent" "dev" {
 
   metadata {
     display_name = "CPU Usage"
-    key          = "cpu"
-    script       = <<EOT
-    vmstat | awk 'FNR==3 {printf "%2.0f%%", $13+$14+$16}'
-    EOT
-    interval     = 1
+    interval     = 10
     timeout      = 1
+    key          = "0_cpu_usage"
+    script       = <<EOT
+      #!/bin/bash
+      # interval in microseconds should be metadata.interval * 1000000
+      interval=10000000
+      ncores=$(nproc)
```

Comment:
Thanks for the insight. Can you check my current implementaion for cgroupv2 only?
I guess it gives a good estimate. I tested with `stress` utility

Reply:
I think you would also need to take into account the number of intervals that have passed since the last check. If you used 100_000 usec of your 100_000 over 1 period of 100_000 usec, that would equal 100% usage. If the same was over 2 periods, it would be 50% usage.

Reply:
yes I use an interval but it has to be set to the interval value of the agent metadata resource when this agent metadata script is called